src/Symfony/Component/DependencyInjection/Loader/Configurator/Traits/FactoryTrait.php

```diff
@@ -22,6 +24,10 @@
      */
     final public function factory($factory)
     {
+        if (is_string($factory) && 1 === substr_count($factory, ':')) {
+            throw new InvalidArgumentException('The short factory notation is not supported by the PHP-based DI configuration.');
```

Comment:
In addition, it should also suggest to use `[ref('service'), 'method']` instead.

Reply:
So the final exception will look like `The service:method factory notation is not supported by the PHP-based DI configuration. Use [ref('service'), 'method'] instead.`

Reply:
LGTM, with double quotes around, and resolved/actual id+method, eg:
```
Invalid factory "%s:%s": the `service:method` notation is not available when using PHP-based DI configuration. Use "[ref('%s'), '%s']" instead.
```

Reply:
Static is still allowed using `factory('class::method')`? Or should that notation also be explicit; thus `factory('class', 'method')`.

Reply:
@nicolas-grekas, done